packages/ra-ui-materialui/src/list/DatagridRow.js

```diff
@@ -145,12 +163,19 @@ const DatagridRow = ({
             {expand && expanded && (
                 <TableRow key={`${id}-expand`} id={`${id}-expand`}>
                     <TableCell colSpan={nbColumns}>
-                        {cloneElement(expand, {
-                            record,
-                            basePath,
-                            resource,
-                            id: String(id),
-                        })}
+                        {isValidElement(expand)
```

Comment:
this is a temporary API. I'm not sure yet whether we should support components and elements everywhere, we'll see if profiling forces us to do that often.